src/Utility/CookieCryptTrait.php

```diff
@@ -129,6 +129,11 @@ protected function _decode($value, $encrypt, $key)
         $this->_checkCipher($encrypt);
         $prefix = 'Q2FrZQ==.';
         $value = base64_decode(substr($value, strlen($prefix)));
```

Comment:
It could be checked for valid signature and base64 decoded in strict mode. Should it be changed in 3.x or 4.x?

Reply:
What would the error/result be on malformed data?

Reply:
In all cases of malformed data, cookie should be rejected and result should be the same (currently empty string).

Currently first 9 bytes can be anything, they are removed. Cookie should be rejected with invalid signature.

Currently base64 encoded data can contain characters from outside the base64 alphabet, invalid characters are silently discarded. In strict mode, `base64_decode` would return `false`, and cookie should be rejected.

It is just for stricter validation, it shouldn't make any problem as it was always formatted as `'Q2FrZQ==.' . base64_encode()` from Cake v1 and maybe earlier.

Reply:
Ok. Those improvements sound good to me. I don't see why they couldn't be done in 3.6 as a bugfix.